go/vt/vttablet/tabletmanager/vreplication/replicator_plan.go

```diff
@@ -646,6 +654,24 @@ func appendFromRow(pq *sqlparser.ParsedQuery, buf *bytes2.Buffer, fields []*quer
 				buf.WriteString(sqltypes.NullStr)
 			} else {
 				vv := sqltypes.MakeTrusted(typ, row.Values[col.offset:col.offset+col.length])
+
+				if conversion, ok := tp.ConvertCharset[col.field.Name]; ok && col.length >= 0 {
+					// Non-null string value, for which we have a charset conversion instruction
+					fromCollation := tp.CollationEnv.DefaultCollationForCharset(conversion.FromCharset)
+					if fromCollation == collations.Unknown {
+						return vterrors.Errorf(vtrpcpb.Code_INVALID_ARGUMENT, "Character set %s not supported for column %s", conversion.FromCharset, col.field.Name)
```

Comment:
nit, but errors aren't supposed to be capitalized (due to wrapping)

Reply:
Fixed! One place where I did leave the message capitalized is in `"Incorrect string value"` - this string mimics the error message MySQL would have given for the equivalent SQL `CONVERT(...)` function, and I think we should keep this as it promotes consistency.